lib/rubocop/cop/rails/save_bang.rb

```diff
@@ -221,15 +221,20 @@ def check_used_in_conditional(node)
           true
         end
 
-        def conditional?(node) # rubocop:disable Metrics/CyclomaticComplexity
+        def in_boolean_expr?(node)
```

Comment:
IMHO, the method name may remain `conditional?`.

Reply:
I think the `conditional?` name is confusing since a ["conditional node"](https://www.rubydoc.info/gems/rubocop/RuboCop/AST/ConditionalNode), according to the Rubocop AST, is one that represents an `if`, `while`, `until`, or other statements that have a body and a condition.

This method doesn't test to see if the node is a conditional node -- it tests to see if the node is _in the condition of_ a conditional node, or if it's in a compound boolean statement.

I'll change it to `in_condition_or_compound_boolean?` -- of course please comment again if you disagree with the new name

Reply:
I get it. Thanks for the explanation!